include/deal.II/matrix_free/fe_evaluation.h

```diff
@@ -3155,31 +3185,31 @@ namespace internal
             typename GlobalVectorType,
             typename FU>
   inline void
-  process_cell_data(
+  process_cell_or_face_data(
     const FEEvaluationBase<dim,
                            n_components,
                            Number,
                            is_face,
-                           VectorizedArrayType> &       fe_eval,
-    const MatrixFree<dim, Number, VectorizedArrayType> *matrix_free,
-    GlobalVectorType &                                  array,
-    VectorizedArrayType2 &                              out,
-    const FU &                                          fu)
+                           VectorizedArrayType> &fe_eval,
+    GlobalVectorType &                           array,
+    VectorizedArrayType2 &                       out,
+    const FU &                                   fu,
+    const bool                                   is_cell_data)
   {
-    (void)matrix_free;
-    Assert(matrix_free != nullptr, ExcNotImplemented());
-    AssertDimension(array.size(),
-                    matrix_free->get_task_info().cell_partition_data.back());
-
     // 1) collect ids of cell
-    const auto cells = fe_eval.get_cell_ids();
+    const auto cells =
+      is_cell_data ? fe_eval.get_cell_ids() : fe_eval.get_face_ids();
```

Comment:
Given that the only use of `fe_eval` is to provide either one or the other, which is selected by the function at hand, wouldn't it make more sense to pass in the `std::array` returned by either `fe_eval.get_cell_ids()` in the `true` case or `fe_eval.get_face_ids()` in the `false` case? This allows you to skip `fe_eval`.